### PR TITLE
[REVIEW] Add ShiftLeft (<<), ShiftRight (>>) and ShiftRightUnsigned (>>>) binops.

### PR DESCRIPTION
**Pull request by nvdbaranec**
Closes  https://github.com/rapidsai/cudf/issues/3791

Note : ShiftRightUnsigned is the closest C++ equivalent to Java's  >>>

### CODE REVIEW

File: cpp/include/cudf/binaryop.hpp [hunk 1]
```diff
@@ -28,31 +28,35 @@ namespace experimental {
...
+  BITWISE_OR,             ///< operator |
+  BITWISE_XOR,            ///< operator ^
+  LOGICAL_AND,            ///< operator &&
+  LOGICAL_OR,             ///< operator ||
+  COALESCE,               ///< operator x,y  x is null ? y : x
+  GENERIC_BINARY,         ///< generic binary operator to be generated with input
+                          ///< ptx code
+  SHIFT_LEFT,             ///< operator <<
+  SHIFT_RIGHT,            ///< operator >>
+  SHIFT_RIGHT_UNSIGNED,   ///< operator >>> (from Java)
```
Comment (jrhemstad) [inline cpp/include/cudf/binaryop.hpp]: This needs standalone documentation about what this is.

Comment (jrhemstad) [reply cpp/include/cudf/binaryop.hpp]: Couldn't this be implemented by first casting to unsigned and then doing a right shift?

Comment (nvdbaranec) [reply cpp/include/cudf/binaryop.hpp]: Sure.  Where did you have in mind? Inline with the enum value?

Comment (jrhemstad) [reply cpp/include/cudf/binaryop.hpp]: Yeah, just something like "Cast to unsigned then right shift" or even better, just call it "Logical right shift"

Comment (devavret) [reply cpp/include/cudf/binaryop.hpp]: > Couldn't this be implemented by first casting the column to unsigned and then doing a right shift?

I am ambivalent. We can say this about all JITted binops which are just a couple of casts away from not requiring JIT. Like, we could do a `int32 x + float y = int64 z` with `(int64)((double)x + (double)y)` but we want to save the casts. OTOH, this case is pretty specialized and we **can** remove an entire operation by this cast.

**Pull request merged**